src/AnnotationThread.js

```diff
@@ -308,21 +317,22 @@ class AnnotationThread extends EventEmitter {
             .then(() => {
                 // Ensures that blank highlight comment is also deleted when removing
                 // the last comment on a highlight
-                firstAnnotation = this.annotations[0];
-                canDeleteAnnotation =
-                    firstAnnotation && firstAnnotation.permissions && firstAnnotation.permissions.can_delete;
-                if (util.isPlainHighlight(this.annotations) && canDeleteAnnotation) {
-                    this.api.delete(firstAnnotation.id);
+                if (this.type === TYPES.highlight && this.canDelete) {
+                    this.api.delete(this.id).then(() => {
+                        // Broadcast thread cleanup if needed
+                        if (!this.threadID) {
+                            this.emit(THREAD_EVENT.reset);
+                        } else {
+                            // Broadcast annotation deletion event
+                            this.emit(THREAD_EVENT.delete);
+                        }
+
+                        this.threadID = null;
+                    });
                 }
 
-                // Broadcast thread cleanup if needed
-                firstAnnotation = this.annotations[0];
-                if (!firstAnnotation) {
-                    this.emit(THREAD_EVENT.threadCleanup);
-                } else {
-                    // Broadcast annotation deletion event
-                    this.emit(THREAD_EVENT.delete);
-                }
+                // Broadcast annotation deletion event
+                this.emit(THREAD_EVENT.delete);
```

Comment:
Is it okay to emit the deletion event twice? Here, and above in the promise resolution step?

Reply:
It only gets fired twice when you're deleting the last comment on a highlight which would delete the entire thread as well